akka-remote/src/main/scala/akka/remote/artery/tcp/ssl/SslManagersProvider.scala

```diff
@@ -0,0 +1,66 @@
+package akka.remote.artery.tcp.ssl
+
+import java.io.FileInputStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.security.KeyStore
+import java.security.cert.Certificate
+import java.security.cert.X509Certificate
+
+import com.typesafe.config.Config
+import javax.net.ssl.KeyManager
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.TrustManager
+import javax.net.ssl.TrustManagerFactory
+
+import scala.util.Try
+
+trait SslManagersProvider {
+  def trustManagers: Array[TrustManager]
+  def keyManagers: Array[KeyManager]
+
+  val caCertificate: Certificate
+  val peerCertificate: X509Certificate
```

Comment:
I've added these two to the trait so any certificate reader is required to provide direct access to each of these two certificates. I don't think adding `caCertificate` is a good idea, instead, I think 

```
  val caCertificates: Array[Certificate]
  val peerCertificate: X509Certificate
```

would probably be better.